### PR TITLE
feat: Allow SSE on AWS S3 Buckets

### PR DESCRIPTION
**Pull request by therynamo**
Signed-off-by: therynamo <theryn.groetken@gemini.com>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

The below screenshot is the integration run through the techdocs CLI - where there is a PR to add this functionality there too. https://github.com/backstage/techdocs-cli/pull/193

techdocs-aws-sse

### CODE REVIEW

**Comment by Xantier** [comment]
Few typos according to `vale` on the changeset but other than that this looks good.

**Comment by therynamo** [comment]
Hey @Xantier - thanks for the quick review. I changed the abbreviations to their full words - so the checks should pass now. Let me know if there is anything else I can do!

File: docs/features/techdocs/configuration.md [hunk 1]
```diff
@@ -106,6 +106,11 @@ techdocs:
       # This allows providers like LocalStack, Minio and Wasabi (and possibly others) to be used to host tech docs.
       s3ForcePathStyle: false
 
+      # (Optional) AWS Server Side Encryption
+      # If not set, encrypted buckets will fail to publish.
+      # https://docs.aws.amazon.com/AmazonS3/latest/userguide/specifying-s3-encryption.html
+      sse: 'aws:kms' # or AES256
```
Comment (iamEAP) [inline docs/features/techdocs/configuration.md:112]: Thanks for getting this documented!  Would you mind also adding this type to the [plugin-techdocs-backend config schema](https://github.com/backstage/backstage/blob/master/plugins/techdocs-backend/config.d.ts#L73)?  Specifying type there as `'aws:kms' | 'AES256'` would also be nice.

File: .changeset/orange-cougars-relax.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/techdocs-common': patch
```
Comment (iamEAP) [inline .changeset/orange-cougars-relax.md:2]: When you add that type to the backend plugin, you'll also want to add `'@backstage/plugin-techdocs-backend': patch` here too.

While you're here: if you rename the file with a `techdocs-` prefix, we can likely get this in a little more quickly!

**Review by iamEAP — COMMENTED** [review]
Thanks @therynamo!  This looks good from my POV.  One comment on config schema, then happy to get this in.

**Comment by therynamo** [comment]
@iamEAP - Thanks for the review - sorry it took so long to get back to you on this. You know how it goes with work and OSS and such. I appreciate your patience! 

So I changed the name from `orange-cougars-relax.md` to `techdocs-orange-cougars-relax.md` - is that what you were asking? 

Also - I updated the types and added a default value comment as well. 

Let me know if there is anything else I need to do! Looking forward to being able to use encrypted buckets 🥳

**Review by iamEAP — APPROVED** [review]
Thanks for the contrib @therynamo!  Just had one little cleanup merge on top of yours so we can get this in this week.

**Pull request merged**